File: app/src/main/java/com/kelsos/mbrc/dao/views/PlaylistTrackView.java
package com.kelsos.mbrc.dao.views;

import com.kelsos.mbrc.RemoteDatabase;
import com.kelsos.mbrc.dao.PlaylistTrackDao;
import com.kelsos.mbrc.dao.PlaylistTrackDao_Table;
import com.kelsos.mbrc.dao.PlaylistTrackInfoDao;
import com.kelsos.mbrc.dao.PlaylistTrackInfoDao_Table;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ModelView;
import com.raizlabs.android.dbflow.annotation.ModelViewQuery;
import com.raizlabs.android.dbflow.sql.Query;
import com.raizlabs.android.dbflow.sql.language.Join;
import com.raizlabs.android.dbflow.sql.language.OrderBy;
import com.raizlabs.android.dbflow.sql.language.SQLite;
import com.raizlabs.android.dbflow.structure.BaseModelView;

@ModelView(database = RemoteDatabase.class, name = "playlist_track_view") public class PlaylistTrackView
    extends BaseModelView<PlaylistTrackDao> {
  @ModelViewQuery public static final Query QUERY = SQLite.select(PlaylistTrackDao_Table.id.withTable(),
      PlaylistTrackDao_Table.playlist_id.withTable(),
      PlaylistTrackDao_Table.position.withTable(),
      PlaylistTrackInfoDao_Table.path.withTable(),
      PlaylistTrackInfoDao_Table.artist.withTable(),
      PlaylistTrackInfoDao_Table.title.withTable())
      .from(PlaylistTrackDao.class)
      .join(PlaylistTrackInfoDao.class, Join.JoinType.INNER)
      .on(PlaylistTrackInfoDao_Table.id.withTable().is(PlaylistTrackDao_Table.track_info_id))
      .orderBy(OrderBy.fromProperty(PlaylistTrackDao_Table.playlist_id).ascending())
      .orderBy(OrderBy.fromProperty(PlaylistTrackDao_Table.position).ascending());

  @Column private long id;
  @Column(name = "playlist_id") private long playlistId;
  @Column private long position;
  @Column private String path;
  @Column private String artist;
  @Column private String title;

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public String getArtist() {
    return artist;
  }

  public void setArtist(String artist) {
    this.artist = artist;
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public long getPosition() {
    return position;
  }

  public void setPosition(long position) {
    this.position = position;
  }

  public long getPlaylistId() {
    return playlistId;
  }

  public void setPlaylistId(long playlistId) {
    this.playlistId = playlistId;
  }

  public long getId() {
    return id;
  }

  public void setId(long id) {
    this.id = id;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/data/views/PlaylistTrackView.kt
package com.kelsos.mbrc.data.views

import com.kelsos.mbrc.data.dao.PlaylistTrack
import com.kelsos.mbrc.data.dao.PlaylistTrack_Table
import com.kelsos.mbrc.data.dao.PlaylistTrackInfo
import com.kelsos.mbrc.data.dao.PlaylistTrackInfo_Table
import com.kelsos.mbrc.data.db.RemoteDatabase
import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ModelView
import com.raizlabs.android.dbflow.annotation.ModelViewQuery
import com.raizlabs.android.dbflow.sql.Query
import com.raizlabs.android.dbflow.sql.language.Join
import com.raizlabs.android.dbflow.sql.language.OrderBy
import com.raizlabs.android.dbflow.sql.language.SQLite
import com.raizlabs.android.dbflow.structure.BaseModelView

@ModelView(database = RemoteDatabase::class,
    name = "playlist_track_view") class PlaylistTrackView : BaseModelView() {

  @Column var id: Long = 0
  @Column(name = "playlist_id") var playlistId: Long = 0
  @Column var position: Long = 0
  @Column var path: String? = null
  @Column var artist: String? = null
  @Column var title: String? = null

  companion object {
    @JvmField @ModelViewQuery val QUERY: Query = SQLite.select(PlaylistTrack_Table.id.withTable(),
        PlaylistTrack_Table.playlist_id.withTable(),
        PlaylistTrack_Table.position.withTable(),
        PlaylistTrackInfo_Table.path.withTable(),
        PlaylistTrackInfo_Table.artist.withTable(),
        PlaylistTrackInfo_Table.title.withTable())
        .from(PlaylistTrack::class.java)
        .join(PlaylistTrackInfo::class.java, Join.JoinType.INNER)
        .on(PlaylistTrackInfo_Table.id.withTable().`is`(PlaylistTrack_Table.track_info_id))
        .orderBy(OrderBy.fromProperty(PlaylistTrack_Table.playlist_id).ascending())
        .orderBy(OrderBy.fromProperty(PlaylistTrack_Table.position).ascending())
  }
}
